aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -2014,7 +2097,15 @@ static Tensor _matmul_impl(
                    : tensor1.unsqueeze(0).mm(tensor2).squeeze_(0);
   } else if (dim_tensor1 == 2 && dim_tensor2 == 2) {
     return has_out ? at::mm_out(out, tensor1, tensor2) : tensor1.mm(tensor2);
-  } else if (should_fold(tensor1, tensor2)) {
+  } else if (should_fold(tensor1, tensor2, key, need_profiling, autotune)) {
+    std::chrono::time_point<std::chrono::high_resolution_clock> start;
```

Comment:
A much cleaner way to implement this would be via a RAII object. This way youñ won't need to repeat the code in both branches, and you won't need to refactor the else branch.